client/components/banner/index.jsx

```diff
@@ -0,0 +1,229 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import {
+	noop,
+	size,
+} from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PLAN_PERSONAL,
+	PLAN_PREMIUM,
+	PLAN_BUSINESS,
+} from 'lib/plans/constants';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { getSelectedSiteSlug } from 'state/ui/selectors';
+import { getValidFeatureKeys } from 'lib/plans';
+import Button from 'components/button';
+import Card from 'components/card';
+import DismissibleCard from 'blocks/dismissible-card';
+import Gridicon from 'components/gridicon';
+import PlanIcon from 'components/plans/plan-icon';
+import PlanPrice from 'my-sites/plan-price';
+
+class Banner extends Component {
+
+	static propTypes = {
+		callToAction: PropTypes.string,
+		className: PropTypes.string,
+		description: PropTypes.string,
+		dismissPreferenceName: PropTypes.string,
+		dismissTemporary: PropTypes.bool,
+		event: PropTypes.string,
+		feature: PropTypes.oneOf( [ ...getValidFeatureKeys() ] ),
+		href: PropTypes.string,
+		icon: PropTypes.string,
+		list: PropTypes.arrayOf( PropTypes.string ),
+		onClick: PropTypes.func,
+		plan: PropTypes.string,
+		price: PropTypes.oneOfType( [ PropTypes.number, PropTypes.arrayOf( PropTypes.number ) ] ),
+		siteSlug: PropTypes.string,
+		title: PropTypes.string.isRequired,
+	}
+
+	static defaultProps = {
+		dismissTemporary: false,
+		onClick: noop,
+	}
```

Comment:
ESLint does not warn yet, but static property and instance property initializers need semicolons.

https://github.com/tc39/proposal-class-public-fields/pull/45/files
https://tc39.github.io/proposal-class-public-fields/